ansible/roles/elastic_beats/tasks/main.yml

```diff
@@ -27,16 +34,25 @@
   ansible.builtin.include_vars:
     file: core.yml
 
-- name: Load platform input config
+- name: Load tenderdash input config
   ansible.builtin.include_vars:
-    file: platform.yml
-  # We make sure that we running it on a node with platform started
+    file: tenderdash.yml
   when: tenderdash_host_info.containers | length > 0
 
-- name: Merge input configs
+- name: Load drive input config
+  ansible.builtin.include_vars:
+    file: drive.yml
+  when: drive_host_info.containers | length > 0
+
+- name: Merge drive and tenderdash input configs
```

Comment:
You can define empty definitions at the top. Then conditionally overwrite each one with actual data and bellow merge all three without any conditions

Reply:
This won't work because `set_fact` has higher precedence than `include_vars`, so the variable definition will still be empty when we try to merge.

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#understanding-variable-precedence